core/src/main/java/org/elasticsearch/search/highlight/CustomQueryScorer.java

```diff
@@ -87,6 +88,12 @@ protected void extractUnknownQuery(Query query,
             }
         }
 
+        protected void extract(Query query, float boost, Map<String, WeightedSpanTerm> terms) throws IOException {
+            // skip all geo queries, see https://issues.apache.org/jira/browse/LUCENE-7293 and
+            // https://github.com/elastic/elasticsearch/issues/17537
+            if (query instanceof GeoPointInBBoxQuery == false) {
```

Comment:
Sorry, nevermind: both of those queries also subclass `GeoPoinInBBoxQuery`, so this check is sufficient!